azure-kusto-data/azure/kusto/data/security.py

```diff
@@ -212,7 +212,13 @@ def get_token_from_msi(self) -> dict:
 
 
 def _get_header_from_dict(token: dict):
-    return _get_header(token[TokenResponseFields.TOKEN_TYPE], token[TokenResponseFields.ACCESS_TOKEN])
+    if TokenResponseFields.TOKEN_TYPE in token:
+        return _get_header(token[TokenResponseFields.TOKEN_TYPE], token[TokenResponseFields.ACCESS_TOKEN])
+    elif OAuth2ResponseParameters.TOKEN_TYPE in token:
+        # Assume OAuth2 format (e.g. MSI Token)
+        return _get_header(token[OAuth2ResponseParameters.TOKEN_TYPE], token[OAuth2ResponseParameters.ACCESS_TOKEN])
+    else:
+        raise KustoClientError("Unexpected token format!")
```

Comment:
hmm, call to action? what should the client do in this case?

Reply:
hmm,
I did not want to leave the OAuth2 format as an 'else' option.
The exception is collected and re-thrown to the user.

Do you mean that the exception description is not detailed enough?
I can change it to "Unexpected token format. Unable to locate token type property."

Reply:
I think the text is too technical. Thinking from the users's perspective - how will he know what to do? Not sure the user will get that this is an auth problem. perhaps have a `KustoAuthError` with some extra details regarding why we couldn't locate the token?

Reply:
There's nothing he can do but contact the developers (unless he is somehow constructing the token by hand).
That said, I'll be more specific regarding the missing jwt property names.

Reply:
P.S. this exception is caught and converted to KustoAuthenticationError by acquire_authorization_header()